### DISCUSSION TITLE
"Set active connection" and "Set active schema" are not available in menu or toolbar.

### DISCUSSION THREAD

**Discussion by Kamurai** [dbeaver/dbeaver#15781] [Q&A]
"Set active connection" and "Set active schema" are not available in menu or toolbar.  For those familiar, it was two drop boxes in the toolbar.

 

This is not present in 6.0.5 which I've been using the better part of a year now.

I've done quite of bit of researching and this doesn't seem to have been noted since 6.0.3, which I was forced to use in a different environment.  Having it upgraded to the current 22.0.0, the problem seems to have returned.

[Imgur](https://i.imgur.com/66cnRTd.png)

 

The user guide seems to think it is an available option, under "Active Database/Schema Selection" :

https://dbeaver.com/wp-content/uploads/wikidocs_archive/dbeaver_v_21_2.pdf

 

#### System information:

- Operating system (distribution) and version: Windows 10, and Windows Azure (both home and work)

- DBeaver version: 22.0.0 (and 6.0.3)

- Additional extensions: n/a

 

#### Connection specification: n/a

- Database name and version

- Driver name

- Do you use tunnels or proxies (SSH, SOCKS, etc)?

 

#### Describe the problem you're observing:

The "set active database connection" and "set active schema" drop boxes are not present.

ctrl+9 and ctrl+0 do not work.

The "set active database connection" and "set active schema" are not available under SQL Editor.

The are not available to be enabled in Custom Perspective:

[Imgur](https://i.imgur.com/obSfIei.png)

#### Steps to reproduce, if exist:

Start program, drop boxes do not display.

Press ctrl+9 or ctrl+0, nothing happens.

Click SQL Editor, options are not available.

Click Window, Custom Perspective, attempt to check the checkbox for "DataSource Selector"

error occurs.  Clicking "Yes" does nothing.

 

#### Include any warning/errors/backtraces from the logs

!ENTRY org.eclipse.ui 4 4 2022-03-07 16:42:39.330

!MESSAGE Plug-in 'org.eclipse.team.ui' contributed an invalid Menu Extension (Path: 'project' is invalid): org.eclipse.team.ui.ConfigureProject

 

!ENTRY org.eclipse.ui 4 4 2022-03-07 16:42:39.331

!MESSAGE Plug-in 'org.eclipse.team.ui' contributed an invalid Menu Extension (Path: 'project' is invalid): org.eclipse.team.ui.ApplyPatchAction

 

!ENTRY org.eclipse.ui 4 4 2022-03-07 16:42:39.333

!MESSAGE Plug-in 'org.eclipse.ui.ide' contributed an invalid Menu Extension (Path: 'navigate/goTo' is invalid): goToResource

**Comment by LonwoLonwo** [comment]
Hello @Kamurai 

What database do you use?

**Comment by Kamurai** [comment]
> Hello @Kamurai 
> 
> What database do you use?

DB2 and SQL Server, but it is irrelevant as my issue occurs before DB selection.

**Comment by LonwoLonwo** [comment]
It works fine for me if any SQL Editor is already open in DBeaver.

Do you have opened the SQL Editor tab in DBeaver?

![2022-03-09 11_00_15-DBeaver Ultimate 22 0 0 - _postgres 13 3_Script-81](https://user-images.githubusercontent.com/45152336/157398073-6ea11984-bd1f-47b9-a7af-18d1a0842afb.png)

**Comment by Kamurai** [comment]
> It works fine for me if any SQL Editor is already open in DBeaver.
> 
> Do you have opened the SQL Editor tab in DBeaver?
> 
> ![2022-03-09 11_00_15-DBeaver Ultimate 22 0 0 - _postgres 13 3_Script-81](https://user-images.githubusercontent.com/45152336/157398073-6ea11984-bd1f-47b9-a7af-18d1a0842afb.png)
> 
> 

I had not opened a new script beforehand, doing so seems to display it correctly.

Thank you, I'm glad this isn't a problem.

**Comment by LonwoLonwo** [comment]
Thanks for the quick response and investigation @Kamurai
